public/testem/testem_client.js

```diff
@@ -186,6 +186,28 @@ var Testem = {
     }
     this.evtHandlers[evt].push(callback);
   },
+  once: function(evt, callback) {
+    if (!this.evtHandlers) {
+      this.evtHandlers = {};
+    }
+    if (!this.evtHandlers[evt]) {
+      this.evtHandlers[evt] = [];
+    }
+    const callbackIndex = this.evtHandlers[evt].length;
+    let wrapper = function(){
+      callback(...arguments);
+      // remove the callback after it is completed
+      this.evtHandlers[evt].splice(callbackIndex,1);
+    };
+    this.evtHandlers[evt].unshift(wrapper);
```

Comment:
whats the reason we use `unshift` here again?

Reply:
we are using `unshift` because we want to prevent removing other events that are added while handling `evt`. Adding `evt` to the front and removing after handling can prevent the case.

Reply:
I gave it some thought:
1. lets say `this.evtHandlers['someEvent'] already has 2 event in its array`. We call `once('someEvent', callback)` and then the event gets added to the front of the list of handlers, now the order of the events are not sequential. This could come as a surprise to the user.  eg: 
`this.evtHandlers['someEvent']`  has  `[event1, event2]`
we call `once('someEvent', event3) where event3 depends on something from event1
now `this.evtHandlers['someEvent']`  looks like  `[event3, event1, event2] `
so when line 171 of emit() is executed:
```
if (this.evtHandlers && this.evtHandlers[evt]) {
      var handlers = this.evtHandlers[evt];
      for (var j = 0; j < handlers.length; j++) {
        var handler = handlers[j];
        handler.apply(this, argsWithoutFirst);
      }
    }
``` 
event3 could throw an error since it was expecting something to be done first in event1.

2. removing the event once the event is complete, seems unnecessary, and might cause unexpected problems:
`this.evtHandlers['someEvent']`  has  `[e1, e2, e3]`
From the emit code, we see that this.evtHandlers uses a loop to go through the array.
At index = 0, if e1 was added with `once()` and once its called, removes itself, now the array looks like `[e2, e3]`
Now the loops goes to index=1, now we run e3, but we did not run e2.

I suggest two things:
1. use `push` instead of `unshift` so the order of the events are consistent
2. instead of doing splice() to remove the callback, we can keep track of the index using an array, then removing the events after the `if` block on line 171